src/pca_gophish.yml

```diff
@@ -48,6 +48,32 @@
             regexp: "published: 3380$"
             line: "        published: 80"
             state: present
+        # I was hoping to override the default published postfix ports for
+        # target (internal container) ports 25 and 587 by including our
+        # preferred published ports in docker-compose.production.yml.
+        # However, when Docker Compose merges the contents of the default
+        # docker-compose.yml with the contents of
+        # docker-compose.production.yml, it does not intelligently overwrite
+        # the published ports.  For example, instead of having target
+        # port 25 published to only host port 25 (as specified in
+        # docker-compose.production.yml), we end up with target port 25
+        # published to BOTH host port 25 (from docker-compose.production.yml)
+        # and host port 1025 (from docker-compose.yml).
+        # That is why the next step removes the default published ports from
+        # docker-compose.yml so that only the ports specified in
+        # docker-compose.production.yml will be published.
+        - name: Remove default postfix published ports
+          ansible.builtin.replace:
+            path: "{{ pca_gophish_composition_path }}/docker-compose.yml"
+            regexp: "^\\s*ports:\\s*$\n\
+            ^\\s*-\\s*target:\\s*['\"]25['\"]\\s*$\n\
+            ^\\s*published:\\s*['\"]1025['\"]\\s*$\n\
+            ^\\s*protocol:\\s*tcp\\s*$\n\
+            ^\\s*mode:\\s*host\\s*$\n\
+            ^\\s*-\\s* target:\\s*['\"]587['\"]\\s*$\n\
+            ^\\s*published:\\s*['\"]1587['\"]\\s*$\n\
+            ^\\s*protocol:\\s*tcp\\s*$\n\
+            ^\\s*mode:\\s*host\\s*$\n"
```

Comment:
This is pretty gnarly.  What do you think about creating an issue for figuring out a better way to do this?

Reply:
I agree that it's a nasty regex.  I looked at [`ansible.builtin.blockinfile`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/blockinfile_module.html), but that would require us to put some marker lines around the text block that we want to modify, so I didn't love that solution.

If you think it's cleaner, I can switch to overwriting the entire `docker-compose.yml` file with one customized for this AMI, which would also mean that we could get rid of `docker-compose.production.yml`.  I didn't go that way originally because it means that whenever [`cisagov/pca-gophish-composition`/docker-compose.yml](https://github.com/cisagov/pca-gophish-composition/blob/develop/docker-compose.yml) changes, we would also need to make similar updates in this repo, but maybe that's not so terrible compared to this regex.  What do you say @jsf9k?

Reply:
I thought about those options too, but I don't really like any of them.  Ansible doesn't have a built-in module to manipulate YAML files (which is really what we need here), so I thought of these options:
1. Convert the YAML to JSON, use `ansible.builtin.command` to run `jq`, then convert the JSON back to YAML.
2. Install and use [`yq`](https://kislyuk.github.io/yq/) via `ansible.builtin.command`.
3. Figure out a way to use the `from_yaml`, `to_json`, `to_yaml`, and `json_query` Ansible filters to achieve this.
4. Use [jsonpatch](https://github.com/particledecay/ansible-jsonpatch).

I'm fine with punting for now, so long as there is an issue to remind us to revisit this.

Reply:
Sounds good to me- I made issue #42 for this.

BTW, this may also be something that a future version of docker-compose can help us overcome:
- https://github.com/docker/compose/issues/2260
- https://github.com/docker/compose/issues/8276

Reply:
[This sort of thang](https://github.com/docker/compose/issues/2260#issuecomment-221050836) might be an OK solution as well.

You'll want to add a (short) paragraph to the comment before this task, stating that the regex is ugly and mentioning the issue you just created.  There is a line in the checklist in the PR template for this sort of thing that you will also need to re-add.

Reply:
> You'll want to add a (short) paragraph to the comment before this task, stating that the regex is ugly and mentioning the issue you just created. There is a line in the checklist in the PR template for this sort of thing that you will also need to re-add.

Good call - see https://github.com/cisagov/pca-gophish-composition-packer/pull/39/commits/a305491ebbd3382c77ad30127d72270117613f6f.